### PR TITLE
solved task

### PR DESCRIPTION
**Pull request by AVykhovanok**

### CODE REVIEW

File: src/main/java/mate/academy/dao/impl/UserDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        return user;
+    }
+
+    @Override
+    public Optional<User> findByEmail(String email) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createQuery("from User u where u.email = :email", User.class)
+                .setParameter("email", email)
+                .uniqueResultOptional();
+        }
```
Comment (Rommelua) [inline src/main/java/mate/academy/dao/impl/UserDaoImpl.java:42]: image

**Review by Rommelua — CHANGES_REQUESTED** [review]
https://mate-academy.github.io/jv-program-common-mistakes/hibernate/add-user-service/add-user-service-checklist
Before you create PR with your homework, you need to go through the checklist under the task and correct all the points described there. The mentor will not check the work until the checklist points are corrected

File: src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+    @Inject
+    private HashUtil hashUtil;
+
+    public User login(String email, String password) throws AuthenticationException {
```
Comment (Ivan95kos) [inline src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java:20]: do not forget about annotation @Override
```suggestion

    @Override
    public User login(String email, String password) throws AuthenticationException {
```

**Review by Ivan95kos — APPROVED** [review]
Nice

**Pull request closed without merging**